File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxLen = 0;
        unordered_set<char> temp;
        int l = 0;
        for(int r=0; r<s.size(); r++){
            while(temp.contains(s[r])){
                temp.erase(s[l]);
                l++;
            }
            temp.insert(s[r]);
            if(temp.size() > maxLen){
                maxLen = temp.size();
            }
        }
        return maxLen;


    }
};

File: Data Structures & Algorithms/sqrtx/submission-8.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l = 0
        r = x
        while l<r:
            m = (l+r)//2
            if m*m==x:
                return m
            elif m*m < x:
                l=m+1
            else:
                r=m
        if x==0:
            return 0
        elif x==1:
            return 1
        return l-1
            